src/allmydata/scripts/tahoe_grid_manager.py

```diff
@@ -0,0 +1,504 @@
+        """
+        :param name: a user-meaningful name for the server
+        :param public_key: ed25519.VerifyingKey the public-key of the
+            storage provider (e.g. from the contents of node.pubkey
+            for the client)
+        """
+        if name in self._storage_servers:
+            raise KeyError(
+                "Already have a storage server called '{}'".format(name)
+            )
+        ss = _GridManagerStorageServer(name, public_key, None)
+        self._storage_servers[name] = ss
+        return ss
+
+    def remove_storage_server(self, name):
+        """
+        :param name: a user-meaningful name for the server
+        """
+        try:
+            del self._storage_servers[name]
+        except KeyError:
+            raise KeyError(
+                "No storage server called '{}'".format(name)
+            )
+
+    def marshal(self):
+        data = {
+            u"grid_manager_config_version": self._version,
+            u"private_key": self._private_key_bytes.decode('ascii'),
+        }
+        if self._storage_servers:
+            data[u"storage_servers"] = {
+                name: srv.marshal()
+                for name, srv
+                in self._storage_servers.items()
+            }
+        return data
+
+
+def _save_gridmanager_config(file_path, grid_manager):
+    """
+    Writes a Grid Manager configuration.
+
+    :param file_path: a FilePath specifying where to write the config
+        (if None, stdout is used)
+
+    :param grid_manager: a _GridManager instance
+    """
+    data = json.dumps(
+        grid_manager.marshal(),
+        indent=4,
+    )
+
+    if file_path is None:
+        print("{}\n".format(data))
+    else:
+        fileutil.make_dirs(file_path.path, mode=0o700)
+        with file_path.child("config.json").open("w") as f:
+            f.write("{}\n".format(data))
+
+
+def _load_gridmanager_config(gm_config):
+    """
+    Loads a Grid Manager configuration and returns it (a dict) after
+    validating. Exceptions if the config can't be found, or has
+    problems.
+
+    :param gm_config str: "-" (a single dash) for stdin or a filename
+    """
+    fp = None
+    if gm_config.strip() != '-':
+        fp = FilePath(gm_config.strip())
+        if not fp.exists():
+            raise RuntimeError(
+                "No such directory '{}'".format(gm_config)
+            )
+
+    if fp is None:
+        gm = json.load(sys.stdin)
+    else:
+        with fp.child("config.json").open("r") as f:
+            gm = json.load(f)
+
+    try:
+        return _GridManager.from_config(gm, gm_config)
+    except ValueError as e:
+        raise usage.UsageError(str(e))
+
+
+def _show_identity(gridoptions, options):
+    """
+    Output the public-key of a Grid Manager
+    """
+    gm_config = gridoptions['config'].strip()
+    assert gm_config is not None
+
+    gm = _load_gridmanager_config(gm_config)
+    print(gm.public_identity())
```

Comment:
print to `options.parent.parent.parent.parent.stdout`?  (I might have lost track of the correct number of parent attributes to resolve)